include/triton/Dialect/Triton/IR/TritonOps.td

```diff
@@ -420,7 +420,7 @@ def TT_ReduceReturnOp: TT_Op<"reduce.return",
 //
 // External Elementwise op
 //
-def TT_ExtElemwiseOp : TT_Op<"ext_elemwise", [Pure, Elementwise, SameOperandsAndResultShape,
+def TT_ExtElemwiseOp : TT_Op<"ext_elemwise", [Pure, ElementwiseOrNoArg,
```

Comment:
I don't think a timer should be pure since you can't arbitrarily reorder it. Also, it would be nice if the `Elementwise` trait was kept, since that's useful information for optimizations. So, I think this needs to be its own Op.

Reply:
Ha! Yeah you're right about `Pure`. Even for general ExternElementwiseOp we probably can't set the Pure trait.